lib/shortcuts/queue.js

```diff
@@ -102,60 +111,65 @@ class Queue {
         Properties: {
           MessageRetentionPeriod: 1209600,
           VisibilityTimeout: DeadLetterVisibilityTimeout,
-          QueueName: {
-            'Fn::Sub': ['${queue}-dead-letter', { queue: QueueName }]
-          }
+          // The dead-letter queue's type must match the main queue's type.
+          FifoQueue,
```

Comment:
`if (FifoQueue === false)`, then the deploy will fail. `undefined` is fine here, but we should add a check somewhere that converts a user-provided `false` into `undefined`.

Reply:
Ah thanks for the reminder. Added that in f1696ea.